### ISSUE TITLE
Quad MIDI to CV: Segmentation Fault

### ISSUE THREAD

**Issue by Cordoha** [VCVRack/Rack#455]
Ubuntu 16.04
Latest dev buid

Using Quad MIDI to CV **to** 2 Unity Modules **to** 2 Multiples **to** Resonator **to** Audio Interface
Rack segmentation faults using "make debug"

> LD_LIBRARY_PATH=dep/lib gdb -ex run ./Rack
GNU gdb (Ubuntu 7.11.1-0ubuntu1~16.5) 7.11.1
Copyright (C) 2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from ./Rack...done.
Starting program: /home/justin/Rack3/Rack/Rack 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
[info] Current working directory: /home/justin/Rack3/Rack
[info] Global directory: ./
[info] Local directory: ./
[info] Loading plugins from ./plugins
[info] Loaded plugin ./plugins/mscHack/plugin.so
[info] Loaded plugin ./plugins/AmalgamatedHarmonics/plugin.so
[info] Loaded plugin ./plugins/Autodafe/plugin.so
[info] Loaded plugin ./plugins/Fundamental/plugin.so
[info] Loaded plugin ./plugins/MentalModules/plugin.so
[info] Loaded plugin ./plugins/LuckyXXL/plugin.so
[info] Loaded plugin ./plugins/cf/plugin.so
[info] Loaded plugin ./plugins/ML_modules/plugin.so
[info] Loaded plugin ./plugins/LOGinstruments/plugin.so
[info] Loaded plugin ./plugins/BogaudioModules/plugin.so
[info] Loaded plugin ./plugins/NauModular/plugin.so
[info] Loaded plugin ./plugins/ESeries/plugin.so
[info] Loaded plugin ./plugins/LindenbergResearch/plugin.so
[info] Loaded plugin ./plugins/QWelk/plugin.so
[info] Loaded plugin ./plugins/HetrickCV/plugin.so
[info] Loaded plugin ./plugins/JERingMod/plugin.so
[info] Loaded plugin ./plugins/sonusmodular/plugin.so
[info] Loaded plugin ./plugins/DekStop/plugin.so
[info] Loaded plugin ./plugins/Befaco/plugin.so
[info] Loaded plugin ./plugins/JW-Modules/plugin.so
[info] Loaded plugin ./plugins/TreasureFreeLinux/plugin.so
[info] Loaded plugin ./plugins/Grayscale/plugin.so
[info] Loaded plugin ./plugins/Autodafe-Drums/plugin.so
[info] Loaded plugin ./plugins/Simple/plugin.so
[info] Loaded plugin ./plugins/Bidoo/plugin.so
[info] Loaded plugin ./plugins/AudibleInstruments/plugin.so
[info] Loaded plugin ./plugins/KarateSnoopy/plugin.so
[info] Loaded plugin ./plugins/VultModules/plugin.so
[info] Loaded plugin ./plugins/Gratrix/plugin.so
[info] Loaded font ./res/DejaVuSans.ttf
[info] Loading patch ./autosave.vcv
[New Thread 0x7fffe6519700 (LWP 17445)]
[New Thread 0x7fffe6498700 (LWP 17446)]
[Thread 0x7fffe6498700 (LWP 17446) exited]
[Thread 0x7fffe6519700 (LWP 17445) exited]
[info] Loaded SVG ./res/ComponentLibrary/PJ3410.svg
[New Thread 0x7fffe6519700 (LWP 17447)]
[New Thread 0x7fffe6498700 (LWP 17448)]
[Thread 0x7fffe6498700 (LWP 17448) exited]
[Thread 0x7fffe6519700 (LWP 17447) exited]
[debug] Opening audio stream 2
[New Thread 0x7fffe6519700 (LWP 17449)]
[New Thread 0x7fffe6498700 (LWP 17450)]
[Thread 0x7fffe6498700 (LWP 17450) exited]
[Thread 0x7fffe6519700 (LWP 17449) exited]
[New Thread 0x7fffe6519700 (LWP 17451)]
[New Thread 0x7fffe6498700 (LWP 17452)]
[debug] Starting audio stream 2
[New Thread 0x7fffe6405700 (LWP 17453)]
[info] Loaded SVG ./plugins/MentalModules/res/MentalSubMixer.svg
[info] Loaded SVG ./res/ComponentLibrary/SynthTechAlco.svg
[info] Loaded SVG ./res/ComponentLibrary/SynthTechAlco_cap.svg
[info] Loaded SVG ./res/ComponentLibrary/PJ301M.svg
[info] Loaded SVG ./res/ComponentLibrary/RoundBlack.svg
[info] Loaded SVG ./plugins/MentalModules/res/MentalMasterClock.svg
[info] Loaded SVG ./res/ComponentLibrary/LEDButton.svg
[info] Loaded font ./plugins/MentalModules/res/Segment7Standard.ttf
[info] Loaded SVG ./plugins/NauModular/res/Perlin.svg
[info] Loaded SVG ./res/ComponentLibrary/ScrewSilver.svg
[info] Loaded SVG ./res/ComponentLibrary/Davies1900hBlack.svg
[info] Loaded SVG ./plugins/Grayscale/res/Algorhythm.svg
[info] Loaded SVG ./res/ComponentLibrary/PB61303.svg
[info] Loaded image ./plugins/AudibleInstruments/res/Clouds.png
[info] Loaded SVG ./res/ComponentLibrary/Rogan3PSRed.svg
[info] Loaded SVG ./res/ComponentLibrary/Rogan3PSGreen.svg
[info] Loaded SVG ./res/ComponentLibrary/Rogan3PSWhite.svg
[info] Loaded SVG ./res/ComponentLibrary/Rogan1PSRed.svg
[info] Loaded SVG ./res/ComponentLibrary/Rogan1PSGreen.svg
[info] Loaded SVG ./res/ComponentLibrary/Rogan1PSWhite.svg
[info] Loaded SVG ./res/ComponentLibrary/Rogan1PSBlue.svg
[info] Loaded SVG ./res/ComponentLibrary/CKD6_0.svg
[info] Loaded SVG ./res/ComponentLibrary/CKD6_1.svg
[info] Loaded SVG ./res/ComponentLibrary/TL1105_0.svg
[info] Loaded SVG ./res/ComponentLibrary/TL1105_1.svg
[info] Loaded image ./plugins/AudibleInstruments/res/Braids.png
[info] Loaded font ./plugins/AudibleInstruments/res/hdad-segment14-1.002/Segment14.ttf
[info] Loaded SVG ./res/ComponentLibrary/Rogan2SGray.svg
[info] Loaded SVG ./res/ComponentLibrary/Rogan2PSWhite.svg
[info] Loaded SVG ./res/ComponentLibrary/Rogan2PSGreen.svg
[info] Loaded SVG ./res/ComponentLibrary/Rogan2PSRed.svg
[New Thread 0x7fffe4890700 (LWP 17455)]
[info] Loaded SVG ./plugins/Fundamental/res/Unity.svg
[info] Loaded SVG ./res/ComponentLibrary/CKSS_0.svg
[info] Loaded SVG ./res/ComponentLibrary/CKSS_1.svg
[info] Loaded image ./plugins/AudibleInstruments/res/Links.png
[info] Loaded image ./plugins/AudibleInstruments/res/Rings.png
[info] Loaded SVG ./res/ComponentLibrary/Trimpot.svg
[info] Loading settings ./settings.json
[New Thread 0x7fffd99a0700 (LWP 17456)]
[New Thread 0x7fffe408f700 (LWP 17457)]
[Thread 0x7fffe408f700 (LWP 17457) exited]

>Thread 12 "Rack" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7fffd99a0700 (LWP 17456)]
0x00007fffe9d457b6 in rings::Part::RenderModalVoice (this=0x1068490, voice=0, performance_state=..., patch=..., frequency=-4.81642311e+34, 
    filter_cutoff=-1.92656924e+35, size=24) at eurorack/rings/dsp/part.cc:311
311             filter_cutoff * filter_cutoff * 24.0f) / filter_cutoff;

**Comment by AndrewBelt** [comment]
Can you reproduce it consistently? If so, can you make sure to rebuild Audible Instruments and report back whether that fixed it?

**Comment by Cordoha** [comment]
Yes consistently. Also meant, Using Quad MIDI to CV to 2 Unity Modules to 2 Multiples to Resonator **to** Strum's SubMixer **to** Audio Interface

Just rebuilt Audible Instruments. Still a Segmentation Fault. It's when I press two keys on my midi controller.

Would you like a picture of how I have it set up?

**Comment by AndrewBelt** [comment]
Can you narrow down the signals to Rings that is making it crash? Does removing Rings remove the crash?

**Comment by Cordoha** [comment]
Was trying to figure out what "Rings" meant, had to Google/ask in IRC. Definitely the name of the module.

Removed the module, pressed keys, did not crash.

**Comment by AndrewBelt** [comment]
Ah, right, I never think in terms of the "Resonator" fake names.
